dataflux_pytorch/benchmark/checkpointing/multinode/train.py

```diff
@@ -0,0 +1,294 @@
+import argparse
+import os
+import socket
+import time
+import torch
+import statistics
+
+from typing import Generator
+from lightning import Trainer
+from lightning.pytorch.callbacks import ModelCheckpoint
+from lightning.pytorch.demos import (LightningTransformer, Transformer,
+                                     WikiText2)
+import torch.distributed
+from torch.utils.data import DataLoader
+from pathlib import Path
+from google.cloud import storage
+
+from dataflux_core import user_agent
+from dataflux_pytorch.lightning import DatafluxLightningCheckpoint
+from dataflux_pytorch.lightning.gcs_filesystem import GCSDistributedWriter, GCSDistributedReader
+from lightning.pytorch.trainer.states import TrainerFn
+from lightning.pytorch.strategies import FSDPStrategy
+from torch.distributed.fsdp import FullyShardedDataParallel as FSDP
+
+from lightning.pytorch.strategies.fsdp import _METADATA_FILENAME
+from torch.distributed.checkpoint import save, load
+from torch.nn import Module
+
+
+class DatafluxFSDPStrategy(FSDPStrategy):
```

Comment:
Is it possible to import the one from the demo instead of reimplementing it here? That way we ensure they don't get out of sync

Reply:
Done